pkg/minikube/bootstrapper/bsutil/extraconfig.go

```diff
@@ -133,18 +133,18 @@ func defaultOptionsForComponentAndVersion(component string, version semver.Versi
 // newComponentOptions creates a new componentOptions
 func newComponentOptions(opts config.ExtraOptionSlice, version semver.Version, featureGates string, cp config.Node) ([]componentOptions, error) {
 	var kubeadmExtraArgs []componentOptions
-	for _, extraOpt := range opts {
-		if _, ok := componentToKubeadmConfigKey[extraOpt.Component]; !ok {
-			return nil, fmt.Errorf("unknown component %q. valid components are: %v", componentToKubeadmConfigKey, componentToKubeadmConfigKey)
-		}
-	}
-
 	keys := []string{}
 	for k := range componentToKubeadmConfigKey {
 		keys = append(keys, k)
 	}
 	sort.Strings(keys)
 
+	for _, extraOpt := range opts {
```

Comment:
I think it would be better to have a validation in start.go
simmilar to this https://github.com/kubernetes/minikube/blob/2e7d9d88aeea7ff70995b148ee151482aeccdfc5/cmd/minikube/cmd/start.go#L806

we can have a func called
```
validateK8sExtraOptions(....)
```

that would require we make some of the funcs in the bsutil Public